### PR TITLE
Node packaging

### PR DESCRIPTION
**Pull request by rorygarand**
Adds a publish script to send the node sdk to the github package registry.

Related to: https://github.com/googleforgames/agones/issues/679.

### CODE REVIEW

File: build/build-sdk-images/node/publish.sh [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+# limitations under the License.
+
+AGONES_PATH=$1
+RELEASE_VERSION=$2
+
+npm login --registry=https://npm.pkg.github.com
+
+cd $AGONES_PATH/sdks/nodejs
+
+npm version $RELEASE_VERSION
```
Comment (steven-supersolid) [inline build/build-sdk-images/node/publish.sh]: How/when does this script get run?

I think updating the version number should be a careful change and follow semver, so potentially even a manual step. Or are these variables automatically populated and the aim is to keep the package version the same as the release version? If that is the case then what will happen when there are no SDK changes and how do we still comply with semver?

Comment (steven-supersolid) [reply build/build-sdk-images/node/publish.sh]: OK I've read #679 . Perhaps best to first focus on the publishing part and ensure it works and can look at again if we want to do version sync?

Comment (markmandel) [reply build/build-sdk-images/node/publish.sh]: I'm happy to remove the version update. i was going to handle the release part - sounds like a check in the release checklist that we update the version (we can do this by hand?) would work.

So for this PR, we could drop the `npm version $RELEASE_VERSION` command?

**Comment by markmandel** [comment]
Ah, this did actually break a thing!

We have a SDK test here:
https://github.com/markmandel/agones/blob/master/test/sdk/nodejs/testSDKClient.js#L15

So we'll need to change the name there in this PR too for it to pass.

We'll also need to update the nodejs example:

So it's worth noting this is a breaking change, just because the package name has changed.

File: build/build-sdk-images/node/publish.sh (hunk 1 above)
Comment (rorygarand) [reply build/build-sdk-images/node/publish.sh]: Correct. I only added that as a potential way to address an earlier comment from you @markmandel.

@steven-supersolid I believe the idea is to run this from `release.mk` but I was going to leave that part to Mark.

**Comment by rorygarand** [comment]
I was planning on updating the docs, is this PR the best place to do it?

Also, I'll add a fix to the test and remove the `npm version` command.

**Comment by markmandel** [comment]
Yep this PR will be the perfect place - we love having it all in one spot.

File: build/build-sdk-images/node/publish.sh (hunk 1 above)
Comment (steven-supersolid) [reply build/build-sdk-images/node/publish.sh]: Looks good and it seems we keep the SDK version the same as the Agones version?

**Comment by markmandel** [comment]
```
> @ testSDK /go/src/agones.dev/agones/test/sdk/nodejs
> node ./testSDKClient.js

internal/modules/cjs/loader.js:670
    throw err;
    ^

Error: Cannot find module '@googleforgames/agones'
    at Function.Module._resolveFilename (internal/modules/cjs/loader.js:668:15)
    at Function.Module._load (internal/modules/cjs/loader.js:591:27)
    at Module.require (internal/modules/cjs/loader.js:723:19)
    at require (internal/modules/cjs/helpers.js:14:16)
    at Object.<anonymous> (/go/src/agones.dev/agones/test/sdk/nodejs/testSDKClient.js:15:19)
    at Module._compile (internal/modules/cjs/loader.js:816:30)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:827:10)
    at Module.load (internal/modules/cjs/loader.js:685:32)
    at Function.Module._load (internal/modules/cjs/loader.js:620:12)
    at Function.Module.runMain (internal/modules/cjs/loader.js:877:12)
```

Not sure how to solve these problems, but figured I'd find the relevant error message for you.

Let me know if you need help running this test locally.

**Comment by steven-supersolid** [comment]
Oh it won't work until published but we shouldn't publish until the tests all pass :) There is however another way for local testing using npm link.
From the test sdk folder run `npm link <relative path to sdk>`  - this could be added to the script. Please note it will change `package-lock.json` but any changes can be discarded
https://docs.npmjs.com/cli/link.html

File: build/build-sdk-images/node/publish.sh (hunk 1 above)
Comment (rorygarand) [reply build/build-sdk-images/node/publish.sh]: That was the intent... I took out the `npm version` command for now though, we can always add it later if we want.

**Comment by markmandel** [comment]
Passing now! This LGTM - @steven-supersolid any issues on your end, or can I approve and merge?

**Comment by steven-supersolid** [comment]
Oh I thought I already approved. LGTM

**Pull request merged**